mypy/semanal_typeddict.py

```diff
@@ -40,6 +39,7 @@
     require_bool_literal_argument,
 )
 from mypy.state import state
+from mypy.subtypes import is_subtype
```

Comment:
Unfortunately, it is not safe to call `is_subtype()` during semantic analysis (except in `semanal_typeargs.py` which is the last pass, but this is not helpful for you). Some types may not be fully analyzed and this will cause all kinds of weird behaviors.

This kind of checks can be only done during type-checking phase.